source/Halibut/Transport/SecureListener.cs

```diff
@@ -230,14 +226,7 @@ async Task ExecuteRequest(TcpClient client)
                         });
 
                         tcpClientManager.AddActiveClient(thumbprint, client);
-                        // Delegate the open stream to the protocol handler - we no longer own the stream lifetime
```

Comment:
Removing this comment concerns me a lot. From what RobW and I could tell, once the `ExchangeMessages` call returns, we _do_ own the stream, and should be responsible for cleaning it up.